R/with.r

```diff
@@ -128,9 +143,13 @@ set_libpaths <- function(paths) {
   invisible(old)
 }
 
+reset_libpaths <- function(paths) {
```

Comment:
I like the `action` idea. Another `with_` function does the same thing right?

I have no objections to moving these functions into a separate package.